source/comp/markv_codec.cpp

```diff
@@ -586,9 +356,11 @@ class MarkvCodecBase {
   };
 
   explicit MarkvCodecBase(spv_const_context context,
-                          spv_validator_options validator_options)
+                          spv_validator_options validator_options,
+                          const MarkvModel* model)
```

Comment:
Need a comment saying model must be non-null.  Or change it to const ref (even better)

Reply:
Using const ref in tha API. I prefer to use a pointer with an object loaned to a class, as it implicitly suggests lifetime. I.e. it can't be done this way
MarkvEncoder encoder(MarkvModel());
encoder.DoSomething();

Reply:
Ok, so you chose to make it a const pointer.  It would be an improvement to assert that the pointer is not null.

Reply:
I'm a bit confused now.  The new commit deletes the copy and assignment constructors.  So is it still possible to have a dangling reference as you describe?

For example, this looks ok to me, (and passes on an ASAN execution):
```
#include <iostream>

class Foo {
 public:
  explicit Foo(int n) : num_(n) {}
  int num() const { return num_; }

 private:
  Foo(const Foo&) = delete;
  const Foo& operator=(const Foo&) = delete;
  int num_;
};

class Bing {
 public:
  explicit Bing(const Foo& f) : f_(f) {}
  int eval() { return f_.num(); }

 private:
  const Foo& f_;
};

int bar() {
  Bing twelve(Foo(12));
  return twelve.eval() + 3;
}

int main() {
  std::cout << bar() << std::endl;
  return 0;
}
```
In this example Bing holds onto a const reference to a Foo that *looks* like it has expired.  But I think the lifetime of the Foo(12) is extended by Bing's holding onto the reference.

Reply:
I'm not convinced.

C++ 2003 12.2/5
The temporary to which the reference is bound or the temporary that is the complete object to a subobject of which the temporary is bound persists for the lifetime of the reference except as specified below. A temporary bound to a reference member in a constructor’s ctor-initializer (12.6.2) persists until the constructor exits. A temporary bound to a reference parameter in a function call (5.2.2) persists until the completion of the full expression containing the call.

C++ 2011 12.2/5
The temporary to which the reference is bound or the temporary that is the complete object of a subobject to which the reference is bound persists
for the lifetime of the reference except:
(5.1) — A temporary object bound to a reference parameter in a function call (5.2.2) persists until the completion of the full-expression containing the call.
(5.2) — The lifetime of a temporary bound to the returned value in a function return statement (6.6.3) is not extended; the temporary is destroyed at the end of the full-expression in the return statement.
(5.3) — A temporary bound to a reference in a new-initializer (5.3.4) persists until the completion of the full-expression containing the new-initializer.